File: SDKs/amazonaws.com/cognito-idp/2016-04-18/java/lib/src/main/java/openapisdk/models/shared/AdminUpdateDeviceStatusRequest.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * AdminUpdateDeviceStatusRequest
 * The request to update the device status, as an administrator.
**/
public class AdminUpdateDeviceStatusRequest {
    @JsonProperty("DeviceKey")
    public String deviceKey;
    public AdminUpdateDeviceStatusRequest withDeviceKey(String deviceKey) {
        this.deviceKey = deviceKey;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("DeviceRememberedStatus")
    public DeviceRememberedStatusTypeEnum deviceRememberedStatus;
    public AdminUpdateDeviceStatusRequest withDeviceRememberedStatus(DeviceRememberedStatusTypeEnum deviceRememberedStatus) {
        this.deviceRememberedStatus = deviceRememberedStatus;
        return this;
    }
    @JsonProperty("UserPoolId")
    public String userPoolId;
    public AdminUpdateDeviceStatusRequest withUserPoolId(String userPoolId) {
        this.userPoolId = userPoolId;
        return this;
    }
    @JsonProperty("Username")
    public String username;
    public AdminUpdateDeviceStatusRequest withUsername(String username) {
        this.username = username;
        return this;
    }
}

File: SDKs/amazonaws.com/cognito-idp/2016-04-18/python/src/sdk/models/shared/adminupdatedevicestatusrequest.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import devicerememberedstatustype_enum as shared_devicerememberedstatustype_enum


@dataclass_json
@dataclasses.dataclass
class AdminUpdateDeviceStatusRequest:
    r"""AdminUpdateDeviceStatusRequest
    The request to update the device status, as an administrator.
    """
    
    device_key: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('DeviceKey') }})
    user_pool_id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('UserPoolId') }})
    username: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('Username') }})
    device_remembered_status: Optional[shared_devicerememberedstatustype_enum.DeviceRememberedStatusTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('DeviceRememberedStatus') }})
    
